lib/include.js

```diff
@@ -175,6 +175,83 @@ Inclusion.include = function (objects, include, options, cb) {
     cb && cb(err, objects);
   });
 
+  // Find the limit of items for `inq`
+  var inqLimit = 256;
+  if(self.dataSource && self.dataSource.settings &&
+    self.dataSource.settings.inqLimit) {
+    inqLimit = self.dataSource.settings.inqLimit;
+  }
+  /**
+   * Find related items with an array of foreign keys by page
+   * @param model The model class
+   * @param filter The query filter
+   * @param fkName The name of the foreign key property
+   * @param pageSize The size of page
+   * @param options Options
+   * @param cb
+   */
+  function findWithForeignKeysByPage(model, filter, fkName, pageSize, options, cb) {
+    var foreignKeys = [];
+    if (filter.where[fkName]) {
+      foreignKeys = filter.where[fkName].inq;
+    } else if (filter.where.and) {
+      // The inq can be embedded inside 'and: []'
+      for (var j in filter.where.and) {
+        if (filter.where.and[j][fkName] &&
+          Array.isArray(filter.where.and[j][fkName].inq)) {
+          foreignKeys = filter.where.and[j][fkName].inq;
+          break;
+        }
+      }
+    }
+    if (filter.limit || filter.skip || filter.offset) {
+      // Force the find to be performed per FK to honor the pagination
+      pageSize = 1;
+    }
+    var size = foreignKeys.length;
+    if(size > inqLimit && pageSize <= 0) {
+      pageSize = inqLimit;
+    }
+    if (pageSize <= 0) {
+      return model.find(filter, options, cb);
+    }
+
+    var listOfFKs = [];
+
+    for (var i = 0; i < size; i += pageSize) {
+      var end = i + pageSize;
+      if (end > size) {
+        end = size;
+      }
+      listOfFKs.push(foreignKeys.slice(i, end));
+    }
+
+    var items = [];
+    async.each(listOfFKs, function(foreignKeys, done) {
+      var newFilter = {};
+      for (var f in filter) {
+        newFilter[f] = filter[f];
+      }
+      if (filter.where) {
+        newFilter.where = {};
+        for (var w in filter.where) {
+          newFilter.where[w] = filter.where[w];
+        }
+      }
+      newFilter.where[fkName] = {
```

Comment:
Will this support the case of `where:{ and: [...] }`? I mean is `where: {inq:foreignKeys, and:[...]}` a valid one?